backend/battling/views.py

```diff
@@ -54,61 +49,56 @@ def form_valid(self, form):
         else:
             messages.success(self.request, "Battle ended! Check e-mail for results.")
 
-            # creator = Team.objects.filter(battle=battle, trainer=battle.creator.id)
-            # creator_pokemon = PokemonTeam.objects.filter(team=creator[0])
-            # creator_team = [pokemon.pokemon for pokemon in creator_pokemon]
-
-            # opponent = Team.objects.filter(battle=battle, trainer=battle.opponent.id)
-            # opponent_pokemon = PokemonTeam.objects.filter(team=opponent[0])
-            # opponent_team = [pokemon.pokemon for pokemon in opponent_pokemon]
-
             get_battle_winner(battle)
 
-            # send_battle_result(battle, creator_team, opponent_team)
-
         return super().form_valid(form)
 
 
-class DeleteTeam(DeleteView):
-    template_name = "battling/delete_team.html"
+class DeleteBattle(DeleteView):
+    template_name = "battling/delete_battle.html"
     success_url = reverse_lazy("home")
-    queryset = Team.objects.all()
+    queryset = Battle.objects.all()
 
     def get_success_url(self):
         messages.success(self.request, "Battle refused!")
 
         return reverse_lazy("home")
 
 
-class DetailBattle(DetailView):
-    template_name = "battling/battle_detail.html"
+class SettledBattles(ListView):  # pylint: disable=too-many-ancestors
+    template_name = "battling/settled_battles.html"
     model = Battle
 
-    def get_object(self, queryset=None):
-        battle = get_object_or_404(Battle, id=self.kwargs["pk"])
+    queryset = Battle.objects.filter(status="SETTLED")
 
-        return battle
+
+class OnGoingBattles(ListView):  # pylint: disable=too-many-ancestors
+    template_name = "battling/ongoing_battles.html"
+    model = Battle
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["created_battles"] = Battle.objects.filter(status="ONGOING").filter(
+            creator=self.request.user
+        )
+        context["invited_battles"] = Battle.objects.filter(status="ONGOING").filter(
+            opponent=self.request.user
+        )
+        return context
+
+
+class DetailBattle(DetailView):
+    template_name = "battling/battle_detail.html"
+    model = Battle
```

Comment:
These two views have similar templates, do we need two separate views for this?

Reply:
The DetailBattle is to show each pokemon and its attributes, the ongoing battles just show a list of the battles. The context is just so I can separate the battles created from the invited